x-pack/plugins/index_management/__jest__/client_integration/create_enrich_policy/create_enrich_policy.test.tsx

```diff
@@ -0,0 +1,176 @@
+    const { exists, component } = testBed;
+    component.update();
+
+    expect(exists('createEnrichPolicyHeaderContent')).toBe(true);
+    expect(exists('createEnrichPolicyDocumentationLink')).toBe(true);
+  });
+
+  describe('Configuration step', () => {
+    it('Fields have helpers', async () => {
+      const { exists } = testBed;
+
+      expect(exists('typePopoverIcon')).toBe(true);
+      expect(exists('uploadFileLink')).toBe(true);
+      expect(exists('matchAllQueryLink')).toBe(true);
+    });
+
+    it('shows validation errors if form isnt filled', async () => {
+      await testBed.actions.clickNextButton();
+
+      expect(testBed.form.getErrorsMessages()).toHaveLength(3);
+    });
+
+    it('Allows to submit the form when fields are filled', async () => {
+      const { actions } = testBed;
+
+      await testBed.actions.completeCreationStep({});
+
+      expect(actions.isOnFieldSelectionStep()).toBe(true);
+    });
+  });
+
+  describe('Fields selection step', () => {
+    beforeEach(async () => {
+      httpRequestsMockHelpers.setGetFieldsFromIndices(getFieldsFromIndices());
+
+      testBed = await setup(httpSetup);
+      testBed.component.update();
+
+      await testBed.actions.completeCreationStep({});
+    });
+
+    it('shows validation errors if form isnt filled', async () => {
+      await testBed.actions.clickNextButton();
+
+      expect(testBed.form.getErrorsMessages()).toHaveLength(2);
+    });
+
+    it('Allows to submit the form when fields are filled', async () => {
+      const { form, actions } = testBed;
+
+      form.setSelectValue('matchField', 'name');
+      form.setSelectValue('enrichFields', 'email');
+
+      await testBed.actions.clickNextButton();
+
+      expect(actions.isOnCreateStep()).toBe(true);
+    });
+
+    it('When no common fields are returned it shows an error callout', async () => {
+      httpRequestsMockHelpers.setGetFieldsFromIndices({
+        commonFields: [],
+        indices: [],
+      });
+
+      testBed = await setup(httpSetup);
+      testBed.component.update();
+
+      await testBed.actions.completeCreationStep({ indices: 'test-1, test-2' });
+
+      expect(testBed.exists('noCommonFieldsError')).toBe(true);
+    });
+  });
+
+  describe('Creation step', () => {
+    beforeEach(async () => {
+      httpRequestsMockHelpers.setGetFieldsFromIndices(getFieldsFromIndices());
+
+      testBed = await setup(httpSetup);
+      testBed.component.update();
+
+      await testBed.actions.completeCreationStep({});
+      await testBed.actions.completeFieldsSelectionStep();
+    });
+
+    it('Shows CTAs for creating the policy', async () => {
+      const { exists } = testBed;
+
+      expect(exists('createButton')).toBe(true);
+      expect(exists('createAndExecuteButton')).toBe(true);
+    });
+
+    it('Shows policy summary and request', async () => {
+      const { find } = testBed;
+
+      expect(find('enrichPolicySummaryList').html()).toMatchSnapshot();
```

Comment:
Just curious - can you explain why you used a snapshot here?

Reply:
Nice one! I think I was debugging something and forgot about it! I've replaced it with a proper assertion now!